homeassistant/components/telegram_bot/__init__.py

```diff
@@ -450,21 +450,49 @@ def _render_template_attr(data, attribute):
     return True
 
 
-def initialize_bot(p_config):
+def initialize_bot(p_config: dict, hass: HomeAssistant) -> Bot:
     """Initialize telegram bot with proxy support."""
     api_key = p_config.get(CONF_API_KEY)
     proxy_url = p_config.get(CONF_PROXY_URL)
     proxy_params = p_config.get(CONF_PROXY_PARAMS)
 
     if proxy_url is not None:
-        # These have been kept for backwards compatibility, they can actually be stuffed into the URL.
-        # Side note: In the future we should deprecate these and raise a repair issue if we find them here.
-        auth = proxy_params.pop("username"), proxy_params.pop("password")
+        auth = None
+        if proxy_params is None:
+            # CONF_PROXY_PARAMS has been kept for backwards compatibility.
+            proxy_params = {}
+        elif "username" in proxy_params and "password" in proxy_params:
+            # Auth can actually be stuffed into the URL, but the docs have previously
+            # indicated to put them here.
+            auth = proxy_params.pop("username"), proxy_params.pop("password")
+            ir.async_create_issue(
+                hass,
+                DOMAIN,
+                "proxy_params_auth_deprecation",
+                breaks_in_ha_version="2024.6.0",
+                is_persistent=False,
+                is_fixable=False,
+                severity=ir.IssueSeverity.WARNING,
+                translation_key="proxy_params_auth_deprecation",
+                learn_more_url="https://github.com/home-assistant/core/pull/112778",
+            )
+        else:
+            ir.async_create_issue(
+                hass,
+                DOMAIN,
+                "proxy_params_deprecation",
+                breaks_in_ha_version="2024.6.0",
+                is_persistent=False,
+                is_fixable=False,
+                severity=ir.IssueSeverity.WARNING,
+                translation_key="proxy_params_deprecation",
+                learn_more_url="https://github.com/home-assistant/core/pull/112778",
+            )
         proxy = httpx.Proxy(proxy_url, auth=auth, **proxy_params)
         request = HTTPXRequest(connection_pool_size=8, proxy=proxy)
     else:
         request = HTTPXRequest(connection_pool_size=8)
-    return Bot(token=api_key, request=request)
+    return Bot(token=str(api_key), request=request)
```

Comment:
mypy raised an error that `api_key` could be `None`

Reply:
Don't copy `None` to string in that case.

Reply:
```suggestion
    return Bot(token=api_key, request=request)
```